libraries/app/database_api.cpp

```diff
@@ -921,71 +921,127 @@ std::map<std::string, full_account> database_api_impl::get_full_accounts( const
-                    [&acnt] (const force_settlement_object& settle) {
-                       acnt.settle_orders.emplace_back(settle);
-                    });
+      for(auto itr = settle_range.first; itr != settle_range.second; ++itr)
+      {
+         acnt.settle_orders.emplace_back(*itr);
+         if(acnt.settle_orders.size() >= api_limit_get_full_accounts) {
+            acnt.more_data_available.settle_orders = true;
+            break;
+         }
+      }
 
       // get assets issued by user
       auto asset_range = _db.get_index_type<asset_index>().indices().get<by_issuer>().equal_range(account->id);
-      std::for_each(asset_range.first, asset_range.second,
-                    [&acnt] (const asset_object& asset) {
-                       acnt.assets.emplace_back(asset.id);
-                    });
+      for(auto itr = asset_range.first; itr != asset_range.second; ++itr)
+      {
+         acnt.assets.emplace_back(itr->id);
+         if(acnt.assets.size() >= api_limit_get_full_accounts) {
+            acnt.more_data_available.assets = true;
+            break;
+         }
+      }
 
       // get withdraws permissions
-      auto withdraw_range = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().equal_range(account->id);
-      std::for_each(withdraw_range.first, withdraw_range.second,
-                    [&acnt] (const withdraw_permission_object& withdraw) {
-                       acnt.withdraws.emplace_back(withdraw);
-                    });
+      auto withdraw_indices = _db.get_index_type<withdraw_permission_index>().indices();
+      auto withdraw_from_range = withdraw_indices.get<by_from>().equal_range(account->id);
+      for(auto itr = withdraw_from_range.first; itr != withdraw_from_range.second; ++itr)
+      {
+         acnt.withdraws.emplace_back(*itr);
+         if(acnt.withdraws.size() >= api_limit_get_full_accounts) {
+            acnt.more_data_available.withdraws = true;
+            break;
+         }
+      }
+      auto withdraw_authorized_range = withdraw_indices.get<by_authorized>().equal_range(account->id);
+      for(auto itr = withdraw_authorized_range.first; itr != withdraw_authorized_range.second; ++itr)
+      {
+         acnt.withdraws.emplace_back(*itr);
+         if(acnt.withdraws.size() >= api_limit_get_full_accounts) {
+            acnt.more_data_available.withdraws = true;
+            break;
+         }
+      }
 
       // get htlcs
       auto htlc_from_range = _db.get_index_type<htlc_index>().indices().get<by_from_id>().equal_range(account->id);
-      std::for_each(htlc_from_range.first, htlc_from_range.second,
-            [&acnt] (const htlc_object& htlc) {
-               acnt.htlcs.emplace_back(htlc);
-            });
+      for(auto itr = htlc_from_range.first; itr != htlc_from_range.second; ++itr)
+      {
+         acnt.htlcs.emplace_back(*itr);
+         if(acnt.htlcs.size() >= api_limit_get_full_accounts) {
+            acnt.more_data_available.htlcs = true;
```

Comment:
separated the lists and also in withdraws as it will suffer the same issue. https://github.com/bitshares/bitshares-core/pull/1749/commits/c20c3254dcc243ee15f6913446bf772dd8b71f81